DigitalLearningSolutions.Data.Tests/Services/PasswordResetServiceTests.cs

```diff
@@ -1,67 +1,187 @@
         {
             // Given
-            A.CallTo(() => userService.GetUsersByEmailAddress(A<string>._)).Returns((new List<AdminUser>(), new List<DelegateUser>()));
+            A.CallTo(() => userService.GetUsersByEmailAddress(A<string>._))
+                .Returns((new List<AdminUser>(), new List<DelegateUser>()));
 
             // Then
-            Assert.Throws<UserAccountNotFoundException>(() => passwordResetService.GenerateAndSendPasswordResetLink("recipient@example.com", "example.com"));
+            Assert.Throws<UserAccountNotFoundException>(
+                () => passwordResetService.GenerateAndSendPasswordResetLink("recipient@example.com", "example.com"));
         }
 
         [Test]
         public void Trying_to_send_password_reset_sends_email()
         {
+            // Given
+            var emailAddress = "recipient@example.com";
+            var adminUser = Builder<AdminUser>.CreateNew().With(user => user.EmailAddress = emailAddress)
+                .Build();
+
+            A.CallTo(() => userService.GetUsersByEmailAddress(A<string>._))
+                .Returns((new[] { adminUser }.ToList(), new List<DelegateUser>()));
+
             // When
-            passwordResetService.GenerateAndSendPasswordResetLink("recipient@example.com", "example.com");
+            passwordResetService.GenerateAndSendPasswordResetLink(emailAddress, "example.com");
 
             // Then
-            A.CallTo(() =>
-                    emailService.SendEmail(A<Email>.That.Matches(e =>
-                        e.To[0] == "recipient@example.com" &&
-                        e.Cc.IsNullOrEmpty() &&
-                        e.Bcc.IsNullOrEmpty() &&
-                        e.Subject == "Digital Learning Solutions Tracking System Password Reset"))
+            A.CallTo(
+                    () =>
+                        emailService.SendEmail(
+                            A<Email>.That.Matches(
+                                e =>
+                                    e.To[0] == emailAddress &&
+                                    e.Cc.IsNullOrEmpty() &&
+                                    e.Bcc.IsNullOrEmpty() &&
+                                    e.Subject == "Digital Learning Solutions Tracking System Password Reset"))
                 )
                 .MustHaveHappened();
+            ;
+        }
+
+        [Test]
+        public async Task Reset_password_hash_is_invalid_if_125_minutes_old()
+        {
+            // Given
+            var createTime = DateTime.UtcNow;
+
+            var resetPasswordModel = Builder<ResetPassword>.CreateNew()
+                .With(m => m.Id = 1)
+                .With(m => m.PasswordResetDateTime = createTime)
+                .With(m => m.ResetPasswordHash = "Old Hash Brown")
+                .Build();
+            A.CallTo(() => passwordResetDataService.FindAsync(1)).Returns(Task.FromResult(resetPasswordModel));
+
+            var candidate = Builder<DelegateUser>.CreateNew()
```

Comment:
@stellake This is the builder library I mentioned a couple of weeks back - I'm happy to replace these with plain class instances if you think it's better. I don't think the builder saves much code here, though it may be more useful in other tests. Happy to chat about it if it's worth it.